keyring/file.go

```diff
@@ -0,0 +1,155 @@
+package keyring
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/user"
+	"path/filepath"
+	"time"
+
+	jose "github.com/dvsekhvalnov/jose2go"
+	"golang.org/x/crypto/ssh/terminal"
+)
+
+type passwordFunc func(string) (string, error)
+
+func terminalPrompt(prompt string) (string, error) {
+	fmt.Printf("%s: ", prompt)
+	b, err := terminal.ReadPassword(1)
+	if err != nil {
+		return "", err
+	}
+	fmt.Println()
+	return string(b), nil
+}
+
+func init() {
+	supportedBackends[FileBackend] = opener(func(name string) (Keyring, error) {
+		return &fileKeyring{
+			PasswordFunc: terminalPrompt,
+		}, nil
+	})
```

Comment:
It may be better to keep use of the `supportedBackends` in `keyring.go` as the scope of how this works isn't immediately obvious